luna/featurevis/images.py

```diff
@@ -70,3 +70,138 @@ def save_image(img, name=None):
+    :param channels: True for gray images
+
+    :return: A randomly generated image
+    """
+    print('initializing image')
+    # We start from a gray image with some random noise
+    if tf.compat.v1.keras.backend.image_data_format() == 'channels_first':
+        img = tf.random.uniform((1, 3, width, height), dtype=tf.dtypes.float32)
+        shape = [img.shape[0], img.shape[1], img.shape[2], img.shape[3]]
+    else:
+        img = tf.random.uniform((1, width, height, 3), dtype=tf.dtypes.float32)
+        shape = [img.shape[0], img.shape[3], img.shape[1], img.shape[2]]  # [batch, ch, h, w]
+
+    if fft:
+        image_f = fft_image(shape, std=std)
+    else:
+        image_f = np.random.normal(size=shape, scale=std).astype(np.float32)
+
+    if channels:
+        output = tf.nn.sigmoid(image_f)
+    else:
+        output = to_valid_rgb(image_f[..., :3], decorrelate=decorrelate, sigmoid=True)
+
+    return output
+
+
+def fft_image(shape, std=None, decay_power=1):
+    """Image parameterization using 2D Fourier coefficients.
+
+    Args:
+        shape : Image shape
+        sd : standard deviation as noise.
+        decay_power : Defaults to 1.
+
+    Returns:
+        image_t: new image in spatial domain
+    """
+    batch, channels, height, width = shape
+    # real valued fft
+    freqs = rfft2d_freqs(height, width)
+
+    init_val_size = (2, batch, channels) + freqs.shape
+    std = std or 0.01
+    init_val = np.random.normal(size=init_val_size, scale=std).astype(np.float32)
+    spectrum_real_imag_t = tf.Variable(init_val)
+
+    scale = 1.0 / np.maximum(freqs, 1.0 / max(width, height)) ** decay_power
+    scale *= np.sqrt(width * height)
+    spectrum_t = tf.complex(spectrum_real_imag_t[0], spectrum_real_imag_t[1])
+    scaled_spectrum_t = scale * spectrum_t
+
+    # convert the spectrum to spatial domain
+    image_t = tf.transpose(tf.signal.irfft2d(scaled_spectrum_t), (0, 2, 3, 1))
+    image_t = image_t[:batch, :height, :width, :channels]
+    image_t = image_t / 4.0
+    return image_t
+
+
+def rfft2d_freqs(height, width):
+    """computation of 2D frequencies of spectrum
+
+    Args:
+        h : image height
+        w : image width
+
+    Returns:
+        spectrum frequency : 2D spectrum frequencies.
+    """
+    freq_y = np.fft.fftfreq(height)[:, None]
+    # when we have an odd input dimension we need to keep one additional
+    # frequency and later cut off 1 pixel
+    if width % 2 == 1:
+        freq_x = np.fft.fftfreq(width)[: width // 2 + 2]
+    else:
+        freq_x = np.fft.fftfreq(width)[: width // 2 + 1]
+    return np.sqrt(freq_x * freq_x + freq_y * freq_y)
+
+
+color_correlation_svd_sqrt = np.asarray([[0.26, 0.09, 0.02],
+                                         [0.27, 0.00, -0.05],
+                                         [0.27, -0.09, 0.03]]).astype("float32")
+
+max_norm_svd_sqrt = np.max(np.linalg.norm(color_correlation_svd_sqrt, axis=0))
+
+color_mean = [0.48, 0.46, 0.41]
```

Comment:
Are these constants to be used inside of the functions? If so, could you move them to the top and capitalize them?

e.g. COLOR_MEAN

Reply:
Yes they are global variables for now.